src/sentry/tasks/post_process.py

```diff
@@ -564,6 +446,160 @@ def process_snoozes(group):
     return False
 
 
+def process_event(data, group_id):
+    from sentry.eventstore.models import Event
+    from sentry.models import EventDict
+
+    event = Event(
+        project_id=data["project"], event_id=data["event_id"], group_id=group_id, data=data
+    )
+
+    set_current_event_project(event.project_id)
+
+    # Re-bind node data to avoid renormalization. We only want to
+    # renormalize when loading old data from the database.
+    event.data = EventDict(event.data, skip_renormalization=True)
+
+    return event
+
+
+def process_event_group(event):
+    # NOTE: we must pass through the full Event object, and not an
+    # event_id since the Event object may not actually have been stored
+    # in the database due to sampling.
+    from sentry.models.group import get_group_with_redirect
+
+    # Re-bind Group since we're reading the Event object
+    # from cache, which may contain a stale group and project
+    event.group, _ = get_group_with_redirect(event.group_id)
+    event.group_id = event.group.id
+
+    # We fetch buffered updates to group aggregates here and populate them on the Group. This
+    # helps us avoid problems with processing group ignores and alert rules that rely on these
+    # stats.
+    with sentry_sdk.start_span(op="tasks.post_process_group.fetch_buffered_group_stats"):
+        fetch_buffered_group_stats(event.group)
+
+    event.group.project = event.project
+    event.group.project.set_cached_field_value("organization", event.project.organization)
```

Comment:
Would it make sense to differentiate the `process_*` methods into two types:
* mutating input data (`update_*`)
* non-mutating inputs (`process_*`)

I'm always weary of methods that mutates the inputs and I'd be nice to have that explicitly called out when scanning through the code.

Reply:
makes sense. will modify